File: 0795-number-of-subarrays-with-bounded-maximum/0795-number-of-subarrays-with-bounded-maximum.js
/**
 * @param {number[]} nums
 * @param {number} left
 * @param {number} right
 * @return {number}
 */
var numSubarrayBoundedMax = function(nums, left, right) {
  let [ans, lo, hi] = [0, -1, -1];
  
  for (let i = 0; i < nums.length; i++) {
    if (nums[i] > right) lo = i;
    if (nums[i] >= left) hi = i;
    
    ans += hi - lo;
  }
  
  return ans;
};

File: 0795-number-of-subarrays-with-bounded-maximum/0795-number-of-subarrays-with-bounded-maximum.py
class Solution:
    def numSubarrayBoundedMax(self, nums: List[int], left: int, right: int) -> int:
        ans, lo, hi = 0, -1, -1
        for i, num in enumerate(nums):
            if num > right:
                lo = i
            if num >= left:
                hi = i
            ans += hi - lo
        return ans